src/PostgREST/DbStructure/Relation.hs

```diff
@@ -32,45 +29,36 @@ data Relation = Relation
   , relColumns  :: [Column]
   , relFTable   :: Table
   , relFColumns :: [Column]
-  , relType     :: Cardinality
-  , relLink     :: Link -- ^ Constraint on O2M/M2O, Junction for M2M Cardinality
+  , relCard     :: Cardinality
   }
   deriving (Eq, Generic, JSON.ToJSON)
 
+-- | The relationship cardinality
+-- | https://en.wikipedia.org/wiki/Cardinality_(data_modeling)
+-- TODO: missing one-to-one
+data Cardinality
+  = O2M ConstraintName -- ^ one-to-many cardinality
+  | M2O ConstraintName -- ^ many-to-one cardinality
+  | M2M Junction       -- ^ many-to-many cardinality
+  deriving (Eq, Generic, JSON.ToJSON)
+
 type ConstraintName = Text
 
 -- | Junction table on an M2M relationship
-data Link
-  = Constraint
-      { constName :: ConstraintName }
-  | Junction
-      { junTable :: Table
-      , junLink1 :: Link
-      , junCols1 :: [Column]
-      , junLink2 :: Link
-      , junCols2 :: [Column]
-      }
+data Junction = Junction
+  { junTable       :: Table
+  , junConstraint1 :: ConstraintName
+  , junColumns1    :: [Column]
+  , junConstraint2 :: ConstraintName
+  , junColumns2    :: [Column]
```

Comment:
This is something not to worry about - those missing record fields are caused by `RecordWildCard` and will be solved with the coverage overlay later.